content/ytt/docs/latest/data-values-schema-migration-guide.md

```diff
@@ -0,0 +1,201 @@
+
+### Multiple data values files
+For this case, be sure that you are [using multiple data values files](ytt-data-values.md#splitting-data-values-into-multiple-files). 
+
+Given a `ytt` configuration with two data values files:
+```bash
+$ tree .
+.
+├── config.yml
+├── values-1.yml
+└── values-2.yml
+```
+
+`values-1.yml`:
+```yaml
+#@data/values
+---
+key1: myVal
+key2: 8080
+```
+
+`values-2.yml`:
+```yaml
+#@data/values
+---
+key2: 8088
+#@overlay/match missing_ok=True
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+The first step is to combine these two sets of data values into a single set of data values (a union set of all top level keys from the data values files).
+`ytt` can do this for you by providing the`--data-values-inspect` flag to your usual `ytt` command:
+```bash
+$ `ytt -f . --data-values-inspect
+key1: myVal
+key2: 8088
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+You can now copy this output into a yaml file and make it a schema by adding the `#@data/values-schema` annotation to the top of a new document.
+
+`values-schema.yml`:
+```yaml
+#@data/values-schema
+---
+key1: myVal
+key2: 8088
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+
+Now just include the `values-schema.yml` file in your `ytt` invocation instead of the multiple data values files.
+
+### Multiple data values files + Private libraries
+In this scenario, we expect your configuration to depend on a ytt library; outlined in the [library module docs](lang-ref-ytt-library.md).
+
+```bash
+$ tree .
+.
+└── config
+    ├── config.yml 
+    ├── values-1.yml
+    ├── values-2.yml
+    └── _ytt_lib
+        └── lib
+            ├── service.yml
+            └── values.yml
+```
+`config.yml`:
+```yaml
+#@ load("@ytt:data", "data")
+#@ load("@ytt:library", "library")
+#@ load("@ytt:template", "template")
+
+#@ lib = library.get("lib").with_data_values(data.values)
+--- #@ template.replace(lib.eval())
+```
+Using the same `values-1.yml` and `values-2.yml` files from the [multiple data values files schema migration example above](#multiple-data-values-files).
+Migrating to schema happens one library at a time. Let's start with the root library, which includes everything at and below the file system level where the `ytt` invocation was called, not including the `_ytt_lib` folder:
+```bash
+.
+└── config
+    ├── config.yml
+    ├── values-1.yml
+    └── values-2.yml
+```
+As seen in the [previous example](#multiple-data-values-files), `values-1.yml` and `values-2.yml` can be combined and replaced with a single schema file:
+`values-schema.yml`:
+```yaml
+#@data/values-schema
+---
+key1: myVal
+key2: 8088
+key3:
+  host: registry.dev.io
+  port: 8080
+```
+Now we have migrated the root library to use schemas, and the `ytt` invocation will succeed as the same as before, but the private library, "lib" is not yet migrated to schema. 
```

Comment:
The fact that `ytt` will work in a mixed-mode is particularly valuable to know. So much so, I'd recommend it gets its own sentence...

```suggestion
Now we have migrated the root library to use schemas, and the `ytt` invocation will succeed as before. Each library can independently opt-in to using schemas.

In this example, we will migrate all libraries. Here, the private library, "lib" is not yet migrated to schema.
```